pkg/drivers/kic/kic.go

```diff
@@ -357,8 +346,14 @@ func (d *Driver) Stop() error {
 
 	}
 
+	// this is needed only by kic drivers because of the entry-point of the base image overrides the STOP and KILL SIGs
+	// other drivers dont need this
 	if err := killAPIServerProc(d.exec); err != nil {
-		glog.Warningf("couldn't stop kube-apiserver proc: %v", err)
+		glog.Warningf("kill kube-apiserver proc: %v", err)
+	}
+
+	if err := killETCDProc(d.exec); err != nil {
```

Comment:
The driver should absolutely not need to know about etcd by name.  Please remove this so that  it does not paper over a more fundamental driver flaw.

Reply:
done removed boht. i belive in the following up PR to try shut down before stop, we will not have the need to kill either apiserver or ectc so I deleted both of them but this PR should be merged after that other PR that fixes stop